tools/wuget/wuget.cpp

```diff
@@ -57,14 +64,30 @@ int main(int argc, char **argv)
             try
             {
                 filesSeen++;
-                if (doWorkunit || !doManifestInfo)
+                if (doWorkunit || doArchive)
                 {
+                    const char *filename = argv[i];
+                    const char *ext = pathExtension(filename);
                     StringBuffer xml;
-                    if (getWorkunitXMLFromFile(argv[i], xml))
+                    if (ext && strieq(ext, ".xml"))
+                        xml.loadFile(filename, false);
+                    else
+                        getWorkunitXMLFromFile(filename, xml);
+                    if (xml.length())
                     {
                         Owned<ILocalWorkUnit> wu = createLocalWorkUnit(xml);
-                        exportWorkUnitToXML(wu, xml.clear(), true, false, true);
-                        printf("%s\n", xml.str());
+                        if (doWorkunit)
+                        {
+                            exportWorkUnitToXML(wu, xml.clear(), true, false, true);
+                            printf("%s\n", xml.str());
+                        }
+                        if (doArchive)
```

Comment:
can't be true if doWorkunit is true, so could be:
```
else if (doArchive)

Reply:
I think it can, if someone specified -w -a (because they wanted both). May be unusual usage but...

Reply:
it can't at the moment, because it is either or in the parameter passing:
```
            else if (strcmp(argv[i], "-w")==0)
                doWorkunit = true;
            else if (strcmp(argv[i], "-a")==0)
                doArchive = true;
```

Reply:
Which is inside a for loop...

Reply:
sorry yeah misreading.